flink-table-store-core/src/main/java/org/apache/flink/table/store/file/utils/SnapshotFinder.java

```diff
@@ -0,0 +1,106 @@
+public class SnapshotFinder {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SnapshotFinder.class);
+
+    public static final String SNAPSHOT_PREFIX = "snapshot-";
+
+    public static final String EARLIEST = "EARLIEST";
+
+    public static final String LATEST = "LATEST";
+
+    public static Long findLatest(Path snapshotDir) throws IOException {
+        return find(snapshotDir, LATEST, Math::max);
+    }
+
+    public static Long findEarliest(Path snapshotDir) throws IOException {
+        return find(snapshotDir, EARLIEST, Math::min);
+    }
+
+    private static Long find(Path snapshotDir, String hintFile, BinaryOperator<Long> reducer)
+            throws IOException {
+        FileSystem fs = snapshotDir.getFileSystem();
+        if (!fs.exists(snapshotDir)) {
+            LOG.debug("The snapshot director '{}' is not exist.", snapshotDir);
+            return null;
+        }
+
+        Path hint = new Path(snapshotDir, hintFile);
+        try {
+            return Long.parseLong(FileUtils.readFileUtf8(hint));
+        } catch (Exception ignore) {
+            FileStatus[] statuses = fs.listStatus(snapshotDir);
+            if (statuses == null) {
+                throw new RuntimeException(
+                        "The return value is null of the listStatus for the snapshot directory.");
+            }
+
+            Long result = null;
+            for (FileStatus status : statuses) {
+                String fileName = status.getPath().getName();
+                if (fileName.startsWith(SNAPSHOT_PREFIX)) {
+                    try {
+                        long id = Long.parseLong(fileName.substring(SNAPSHOT_PREFIX.length()));
+                        result = result == null ? id : reducer.apply(result, id);
+                    } catch (NumberFormatException e) {
+                        throw new RuntimeException(
+                                "Invalid snapshot file name found " + fileName, e);
+                    }
+                }
+            }
+            return result;
+        }
+    }
+
+    public static void commitLatestHint(Path snapshotDir, long snapshotId) {
+        commitHint(snapshotDir, snapshotId, LATEST);
+    }
+
+    public static void commitEarliestHint(Path snapshotDir, long snapshotId) {
+        commitHint(snapshotDir, snapshotId, EARLIEST);
+    }
+
+    private static void commitHint(Path snapshotDir, long snapshotId, String fileName) {
+        try {
+            FileSystem fs = snapshotDir.getFileSystem();
+            Path hintFile = new Path(snapshotDir, fileName);
+            Path tempFile = new Path(snapshotDir, UUID.randomUUID() + "-" + fileName + ".temp");
+            FileUtils.writeFileUtf8(tempFile, String.valueOf(snapshotId));
+            fs.delete(hintFile, false);
+            fs.rename(tempFile, hintFile);
+        } catch (IOException e) {
+            LOG.warn(String.format("Commit hint file '%s' failed.", fileName), e);
+        }
```

Comment:
Throw directly, so that commit operation can fail property. If you run commit IT cases repeatedly they'll fail.